### DISCUSSION TITLE
Wizzard form/ Funnel

### DISCUSSION THREAD

**Discussion by TOcvfan** [react-hook-form/react-hook-form#9071] [General] [answered]
I'm trying to make a register form on different pages and I saw [this](https://react-hook-form.com/advanced-usage/#WizardFormFunnel) guide. but it didn't work because I'm using a newer version of react-router-dom, than in the guide and some functions are deprecated in the newer version,
And I'm not sure if I should downgrade or how to make it work in the new version.
So I'm wondering how soon guide will be updated..

**Comment by bluebill1049** [comment]
I am not following closely with react-router API change, which API is broken now?

**Reply by TOcvfan** [reply]
it's not broken just updated, just like react-hook-form does 
withRouter is deprecated
and react-router  works fine, but the guide needs to be updated to the newer version

**Comment by leapful** [comment, marked as answer]
@TOcvfan : I think the guide is just an idea about how to manage form values between different routes and pages, it was never a guide to integrate with "react-router-dom" library. In the guide, there is no mention a specific routing library meaning people who uses other routing library can take reference on it to make it work for themself.

If you are working with React Router v5, then my recommendation is to stick with it, changing major version of 3rd party library can be a pitfall on current code base.

Beside that, if you have no legacy code, then just go ahead with React Router v6.

Working demo: https://codesandbox.io/s/react-hook-form-wizard-form-forked-ez3lzx

**Reply by bluebill1049** [reply]
I agree.
